### PR TITLE
Replace linkchecker with cargo-deadlinks

### PR DESCRIPTION
**Pull request by bugadani**
Hi! Thank you for helping out with Embedded Graphics development! Please:

- [ ] Check that you've added passing tests and documentation
- [ ] Add a simulator example(s) where applicable
- [ ] Add a `CHANGELOG.md` entry in the **Unreleased** section under the appropriate heading (**Added**, **Fixed**, etc) and appropriate crate (`embedded-graphics`, `simulator`, `tinytga`, `tinybmp`) if your changes affect the **public API**
- [ ] Run `rustfmt` on the project
- [ ] Run `./build.sh` (Linux/macOS only) and make sure it passes. If you use Windows, check that CI passes once you've opened the PR.

## PR description

Let's see how this goes. One downside may be that deadlinks hasn't been updated for a while, though, so this may or may not be acceptable.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -29,8 +29,4 @@ for crate in "${CRATES[@]}"; do
 done
```
Comment (rfuest) [inline build.sh:29]: It seems that I forgot to remove this file in #410. If you want you can delete it.

Comment (bugadani) [reply build.sh:29]: You mean the whole script?

Comment (rfuest) [reply build.sh:29]: Yes, it was replaced by the `justfile`.

Comment (bugadani) [reply build.sh:29]: Thanks, I'll do that in a different PR.

**Comment by jamwaffles** [comment]
That looks like a _very_ dormant repository which is unfortunate, but it appears the tool does work. This could be a good, platform-agnostic standin until intra-doc links are stabilised. What do you think @rfuest? I'd be happy to move more of the build dependencies into Rust tools.

The CI time has lengthened again, so it might be time to create a custom build image in another PR.

**Comment by rfuest** [comment]
> What do you think @rfuest?

The Python linkchecker seems to be more actively developed but doesn't seem to be in a perfect state either because the Python 3 version needs to be installed from the git repo. So I think it's OK to use `cargo-deadlinks` at the moment.

> The CI time has lengthened again, so it might be time to create a custom build image in another PR.

Yes, I think that's a good idea. On the other side the link check itself seems to be much faster. On my aging system linkcheck takes 41 seconds while deadlinks uses less than 1 second. I hope this improvement doesn't mean that the check isn't as thorough and just means that Rust is far superior to Python :wink:.

**Comment by jamwaffles** [comment]
> The Python linkchecker seems to be more actively developed but doesn't seem to be in a perfect state either

One day, intra-doc links will stabilise and this whole problem will go away :see_no_evil: 

>  I hope this improvement doesn't mean that the check isn't as thorough and just means that Rust is far superior to Python

Hopefully it's the latter :crossed_fingers:

**Comment by bugadani** [comment]
> One day, intra-doc links will stabilise and this whole problem will go away

It's marked for 1.47.0

If something goes bad, this can always be reverted.

**Comment by jamwaffles** [comment]
> It's marked for 1.47.0

TIL. That's exciting!

**Comment by bugadani** [comment]
> > It's marked for 1.47.0
> 
> TIL. That's exciting!

Don't get too excited, someone will have to go through the links :P

**Comment by jamwaffles** [comment]
porting

**Pull request merged**